ORM/ORMFundamentals/Repositories/ProductRepository.cs

```diff
@@ -0,0 +1,44 @@
+﻿using Microsoft.EntityFrameworkCore;
+using ORMFundamentals.Data;
+using ORMFundamentals.Entities;
+
+namespace ORMFundamentals.Repositories;
+
+public class ProductRepository : IProductRepository
+{
+    private readonly AppDbContext _context;
+
+    public ProductRepository(AppDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<IEnumerable<Product>> GetAllProducts()
+    {
+        return await _context.Products.ToListAsync();
```

Comment:
One more important thing to remember about fetching methods in EF - [`AsNoTracking`](https://learn.microsoft.com/en-us/ef/core/querying/tracking). We will discuss it later on our Q&A session, for now, I would just recommend to add overloading or just default parameter to control this flow.